### DISCUSSION TITLE
How to stop eglot from including headers automatically for C/C++ code?

### DISCUSSION THREAD

**Discussion by CDitzel** [joaotavora/eglot#1015] [General]
Its all in the title. At times, auto-completing certain things apparently includes headers automatically at the top of the files. Often this happens unnoticed and causes problems later down the line. 

Is there a command to prevent this functionality?

**Comment by joaotavora** [comment]
You probably need to pass a setting to `clangd`, the language server that -- I presume -- you are using.  To pass settings to servers, use `eglot-server-programs`. or other configuration mechanisms such as `eglot-workspace-configuration`.

**Comment by CDitzel** [comment]
is there an example to look at? =)

**Reply by joaotavora** [reply]
not that I'm aware.  ask the clangd people how their server works and make a example that solves your problem.

**Comment by Eason0210** [comment]
@CDitzel 

I use this setting, and it works for me. Maybe it would be help.

```elisp
(add-to-list 'eglot-server-programs '((c++-mode c-mode) . ("clangd" "--header-insertion=never")))
```
